WordPress/gutenberg
# Refactor: Replace `postId` with `id` in navigation calls across components

Fixes: #67443 

## What?
This PR resolves a bug in the `onNavigateToEntityRecord`, where the `id` parameter was incorrectly referenced as `postId`. This mismatch caused the "Edit" and "Edit Original" buttons to generate URLs containing `undefined`, resulting in broken links for editing individual template parts or patterns.

## How?
The issue was resolved by updating all instances of `onNavigateToEntityRecord` to correctly use the `id` parameter instead of `postId`. This change ensures that valid URLs are generated for editing actions.

## Testing Instructions
1. Activate a block theme.
2. Activate Gutenberg trunk
3. Go to Appearance > Editor. Select a template to edit.
4. Locate or insert any template part in the template.
5. With the template part selected, locate and click on the "Edit" option in the block toolbar.
6. Observe the individual level editing opens up as expected, earlier, it used to produce a warning message.

## Screenshots or screencast 

https://github.com/user-attachments/assets/b3a8581c-879b-46af-a804-cb8edb2101e8
